android/BOINC/app/src/main/java/edu/berkeley/boinc/rpc/RpcClient.java

```diff
@@ -1338,4 +1347,43 @@ public synchronized boolean runBenchmarks() {
             return false;
         }
     }
-}
\ No newline at end of file
+
+    private boolean initBuffersFromSocket()
+    {
+        boolean isLocal = mSocket != null;
```

Comment:
It's safer to pass `isLocal` as a parameter to this function to delegate this decision to the caller function.

Reply:
ok